src/openloco/things/vehicle.h

```diff
@@ -88,7 +88,7 @@ namespace openloco
         thing_id_t next_car_id;     // 0x3A
         uint8_t pad_3C[0x40 - 0x3C];
         uint16_t object_id; // 0x40
-        uint8_t var_42;
+        vehicle_class v_class;
```

Comment:
As this now has a proper type, the loop constructs in `vehicles_menu_mouse_down` and `build_vehicles_menu_mouse_down` could use it as well:

https://github.com/OpenLoco/OpenLoco/blob/0a738d35ab68903ded48739e0b5eeacfd8033c7e/src/openloco/windows/toolbar_top.cpp#L646

https://github.com/OpenLoco/OpenLoco/blob/0a738d35ab68903ded48739e0b5eeacfd8033c7e/src/openloco/windows/toolbar_top.cpp#L720

Reply:
Yes sorry that was my plan. I just didn't have the time this evening.

Reply:
Oh it's totally fine, I'd expected you to say it'd be outside of the scope of this PR, even. :) Just wanted to note that it opened up the possibility.

Reply:
The next question is how best to implement the array. Do i make a map that looks up vehicle type or do I just go for an array and cast the vehicle type to an int.

Reply:
If you make it an `enum` rather than an `enum class` the type should implicitly convert to `uint8_t`.

Reply:
I think strongly typed is good for all the other uses of it though. I think I'll just do a cast.

Reply:
`static_cast` should be fine too, yeah.

Reply:
If we're going for strongly typed, wouldn't a map make more sense?